docs/source/checks/nlp/model_evaluation/plot_metadata_segments_performance.py

```diff
@@ -0,0 +1,107 @@
+   highest feature importance to the model provided (within the features selected for check, if limited).
+
+#. We train multiple simple tree based models, each one is trained using exactly two
+   features (out of the ones selected above) to predict the per sample error calculated before.
+
+#. We convert each of the leafs in each of the trees into a segment and calculate the segment's performance. For the
+   weakest segments detected we also calculate the model's performance on data segments surrounding them.
+"""
+#%%
+# Generate data & model
+# =====================
+
+from deepchecks.nlp.datasets.classification.tweet_emotion import load_data, load_precalculated_predictions
+
+_, test_dataset = load_data(data_format='TextData')
+_, test_probas = load_precalculated_predictions(pred_format='probabilities')
+
+test_dataset.metadata.head(3)
+
+#%%
+# Run the check
+# =============
+#
+# The check has several key parameters (that are all optional) that affect the behavior of the
+# check and especially its output.
+#
+# ``columns / ignore_columns``: Controls which columns should be searched for weak segments. By default,
+# uses all columns.
+#
+# ``alternative_scorer``: Determines the metric to be used as the performance measurement of the model on different
+# segments. It is important to select a metric that is relevant to the data domain and task you are performing.
+#
+# ``segment_minimum_size_ratio``: Determines the minimum size of segments that are of interest. The check is tuned
+# to find the weakest segment regardless of the segment size and so it is recommended to try different configurations
+# of this parameter as larger segments can be of interest even the model performance on them is superior.
+#
+# ``categorical_aggregation_threshold``: By default the check will combine rare categories into a single category called
+# "Other". This parameter determines the frequency threshold for categories to be mapped into to the "other" category.
+#
+# see :class:`API reference <deepchecks.tabular.checks.model_evaluation.WeakSegmentsPerformance>` for more details.
+
+from deepchecks.nlp.checks import MetadataSegmentsPerformance
+from sklearn.metrics import make_scorer, f1_score
+
+scorer = {'f1': make_scorer(f1_score, average='micro')}
+check = MetadataSegmentsPerformance(alternative_scorer=scorer,
+                                    segment_minimum_size_ratio=0.03)
+result = check.run(test_dataset, probabilities=test_probas)
+result.show()
+
+#%%
+# Observe the check's output
+# --------------------------
+#
+# We see in the results that the check indeed found several segments on which the model performance is below average.
+# In the heatmap display we can see model performance on the weakest segments and their environment with respect to the
+# two features that are relevant to the segment. In order to get the full list of weak segments found we will inspect
+# the result.value attribute.
+
+
+result.value['weak_segments_list'].head(3)
```

Comment:
1. Would explicitly mention that in the display we show only "top 3" "lowest performers" but actually the list may be longer and they can be observed etc.... (you did mention that just maybe emphasize that in the display we see only top 3). Accordingly wouldn't limit here to top 3 (but other number) cause otherwise one can assume that we don't have more if they don't read.

Specifically - not necessarily the "top segements" are most interesting since we only do a "go-no go" with segment ratio. So other ones may be more interesting if the have a larger percentage of data.
BTW in your example it's actually like that, see the last weak segment:
![230418 at 00 29 36 screenshot Acrobat](https://user-images.githubusercontent.com/33841818/232614970-62de66bb-dd66-41d2-a3e4-9eac8ebbe1ce.png)
Would be worth mentioning it in "understanding the results". Assuming we use constant seeds along the way (in example here + in building the trees) so that it will reproduce...

Sidenote - a bit confusing that the output dataframe indices are not ordered. (see screenshot)

2. What isn't clear is - how many do we have? what is the threshold by which we decide if there is a weak segment or not? (corresponding to how many weak segment will we have..? or will we always have?)